src/util/SE3.py

```diff
@@ -61,6 +63,18 @@ def from_tf_tree(cls, tf_buffer: tf2_ros.Buffer, parent_frame: str, child_frame:
         result = SE3(position=numpify(tf_msg.translation), rotation=SO3(numpify(tf_msg.rotation)))
         return result
 
+    @classmethod
+    def from_se3_time(cls, tf_buffer: tf2_ros.Buffer, parent_frame: str, child_frame: str) -> Tuple[SE3, Time]:
+        """
+        Ask the TF tree for the time stamp of a frame and return it in a tuple with an
+        SE3 that holds the transform from the parent to the child_frame
+        """
+
+        tf_msg = tf_buffer.lookup_transform(parent_frame, child_frame, rospy.Time()).transform
+        result = SE3(position=numpify(tf_msg.translation), rotation=SO3(numpify(tf_msg.rotation)))
+        time_stamp = tf_buffer.lookup_transform(parent_frame, child_frame, rospy.Time()).header.stamp
```

Comment:
instead of looking up the same tf twice, you can just get the stamp from tf_msg

Reply:
The only reason I decided to look up the tf twice was because the tf_msg stores .transform though I needed to access .header.stamp which are not in transform I don't think.

Reply:
I can just make another variable though so I think that should work.